lib/iris/util.py

```diff
@@ -1171,7 +1171,7 @@ def _handle_dimensional_metadata(
         cube_add_method(new_dm_item, new_dims)
 
     if scalar_coord is not None:
-        scalar_coord = src_cube.coord(scalar_coord)
+        scalar_coord = src_cube.coord(scalar_coord, dim_coords=False)
```

Comment:
@scottrobinson02 Neat use of `dim_coords=False` :+1: 

Note that, the traceback generated here when the `scalar_coord` already exists on the `src_cube` **and** is associated with a cube dimension, is the same traceback when the `src_cube` doesn't contain the provided `scalar_coord`.

This might be a source of confusion to the user. Perhaps it might be clearer to generate different tracebacks for each of these cases i.e., 

1. if the `scalar_coord` already exists on the `src_cube` and is a dimension coordinate, then generate a new exception message to say that the `scalar_coord` cannot be promoted to the new axis as it is already a dimension coordinate of another existing dimension (perhaps even give the cube dimension offset that it covers)
2. otherwise, call `src_cube.coord(scalar_coord, dim_coords=False)` which will either succeed, or generate an exception because the `scalar_coord` genuinely doesn't exist on the `src_cube` i.e., `iris.exceptions.CoordinateNotFoundError: "Expected to find exactly 1 coordinate matching the given 'spam' coordinate's metadata, but found none."`

My point here is that the **same** traceback is generated in each of the following two **different** scenarios:

```python
import iris
from iris.coords import DimCoord
from iris.cube import Cube
from iris.util import new_axis

cube = Cube(1)
spam = DimCoord(1, long_name="spam")
parrot = DimCoord(1, long_name="parrot")

cube.add_aux_coord(spam)

new_cube = new_axis(cube, spam)
new_new_cube = new_axis(new_cube, spam)
```

and

```python
import iris
from iris.coords import DimCoord
from iris.cube import Cube
from iris.util import new_axis

cube = Cube(1)
spam = DimCoord(1, long_name="spam")
parrot = DimCoord(1, long_name="parrot")

cube.add_aux_coord(spam)

new_cube = new_axis(cube, spam)
new_new_cube = new_axis(new_cube, parrot)
```